orttraining/orttraining/test/gradient/gradient_ops_test.cc

```diff
@@ -1207,6 +1207,108 @@ TEST(GradientCheckerTest, SparseSoftmaxCrossEntropyGrad) {
   TestSparseSoftmaxCrossEntropyGrad({2, 3, 2}, "sum");
 }
 
+void TestSoftmaxCrossEntropyLossGrad(const TensorShape& index_shape, const std::string& reduction,
+                                     int64_t ignore_index = 0, int64_t D = 2) {
+  float max_error;
+  GradientChecker<float, float, float> gradient_checker;
+  OpDef op_def{"SoftmaxCrossEntropyLoss", kOnnxDomain, 12};
+  std::function<float(float)> transformer_index = [D](float x) { return std::fmod(std::fabs(x) * 5.0f, D * 1.0f); };
+  std::function<float(float)> transformer_weight = [](float x) { return std::fmod(std::fabs(x), 2.0f); };
+
+  // without weight and ignore_index
+  {
+    std::vector<int64_t> logit_shape(index_shape.GetDims());
```

Comment:
Its not exactly duplicated, there are differences, and if I were to make this a function it will still contain 4 different if-else blocks, I don't strongly feel this is going to help with code maintainability as it just used by this function.

---
In reply to: [408501262](https://github.com/microsoft/onnxruntime/pull/3465#discussion_r408501262) [](ancestors = 408501262)